src/gui/widgets/chatbox.cpp

```diff
@@ -587,20 +587,18 @@ void chatbox::process_message(const ::config& data, bool whisper /*= false*/)
 	if(whisper) {
 		add_whisper_received(sender, message);
 	} else {
-		if (!prefs::get().parse_should_show_lobby_join(sender, message)) return;
-
-		std::string room = data["room"];
+		if(message.size() > 14 && sender == "server" && data["type"] == "error" && std::string_view(message.data(), 12) == "Can't find '") {
+			bool dummy_var = false;
 
-		// Attempt to send to the currently active room first.
-		if(room.empty()) {
-			LOG_LB << "Message without a room from " << sender << ", falling back to active window";
-			room = open_windows_[active_window_].name;
+			close_window_button_callback(message.substr(12, message.size() - 14), dummy_var, dummy_var);
```

Comment:
What is this for?

Reply:
It closes the whisper room using the close button callback.

Reply:
When the server sends the cant find error in response to the whisper message.

Reply:
That seems like a confusing thing to do. Now you cannot see the message you sent at all anymore? What if you want to resend it once the other client is online again? (Unfortunately we don't have history functionality here either for some reason.)

Reply:
Then would it be better If the can't find player error message be displayed in the whisper room? If it is displayed in the lobby or this game room it could be easy to miss when the whisper room is active.

Reply:
I don't like the message parsing but that would be useful. I'd rather extend the message format and add a `original_receiver` attribute to the error message that the client can use to find a fitting tab to display the message.